### PR TITLE
Test all mount related functions

### PR DESCRIPTION
**Pull request by landerweit-phytec**
All new tests need to be run as root.

### CODE REVIEW

**Comment by mschwan-phytec** [comment]
The unit test sporadically fails:
```
1/1 partup:root / mount RUNNING
>>> MALLOC_PERTURB_=88 G_TEST_SRCDIR=/home/schwan/documents/github/phytec/partup/tests G_DEBUG_MESSAGES=all G_TEST_BUILDDIR=/home/schwan/documents/github/phytec/partup/build/tests /home/schwan/documents/github/phytec/partup/build/tests/mount
――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――
TAP version 13
# random seed: R02S6b91ac199a1ba2231876f89fcc1dc5a7
1..6
# Start of mount tests
ok 1 /mount/create_mount_point
ok 2 /mount/mount
ok 3 /mount/umount
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
ok 4 /mount/device_mounted_false
**
ERROR:../tests/mount.c:110:test_device_mounted_true: 'pu_umount(mount_point, &fixture->error)' should be TRUE
not ok /mount/device_mounted_true - ERROR:../tests/mount.c:110:test_device_mounted_true: 'pu_umount(mount_point, &fixture->error)' should be TRUE
Bail out!
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
1/1 partup:root / mount FAIL            0.32s   killed by signal 6 SIGABRT
```
Also, `tests/data/root.ext4` gets modified after running the unit tests. This should be prevented. Maybe the change comes through the mount/umount operation? Looking at the hexdump of `tests/data/root.ext4` before and after running the tests did not reveal any obvious things, though. :thinking:

File: tests/mount.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+{
+    g_autoptr(GError) error = NULL;
+    g_autofree gchar *mount_point = NULL;
+    mount_point = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+
+    const gchar *cmd = g_strdup_printf("mount %s %s", MOUNT_SOURCE, mount_point);
+    gint wait_status;
+
+    g_assert_true(g_spawn_command_line_sync(cmd, NULL, NULL,
```
Comment (mschwan-phytec) [inline tests/mount.c:62]: Check line wrapping.

File: tests/mount.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    gint wait_status;
+
+    mount_point = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+
+    g_assert_true(pu_mount(MOUNT_SOURCE, mount_point, &error));
+    g_assert_no_error(error);
+
+    const gchar *cmd = g_strdup_printf("umount %s", mount_point);
+    g_assert_true(g_spawn_command_line_sync(cmd, NULL, NULL,
```
Comment (mschwan-phytec) [inline tests/mount.c:44]: Check line wrapping.

File: tests/mount.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+static void
+test_umount(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_autofree gchar *mount_point = NULL;
+    mount_point = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+
+    const gchar *cmd = g_strdup_printf("mount %s %s", MOUNT_SOURCE, mount_point);
+    gint wait_status;
```
Comment (mschwan-phytec) [inline tests/mount.c:60]: Please separate declaration and initialization of variables. See also other instances.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]
Some styling issues...

**Comment by landerweit-phytec** [comment]
> The unit test sporadically fails:
> 
> ```
> 1/1 partup:root / mount RUNNING
> >>> MALLOC_PERTURB_=88 G_TEST_SRCDIR=/home/schwan/documents/github/phytec/partup/tests G_DEBUG_MESSAGES=all G_TEST_BUILDDIR=/home/schwan/documents/github/phytec/partup/build/tests /home/schwan/documents/github/phytec/partup/build/tests/mount
> ――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――
> TAP version 13
> # random seed: R02S6b91ac199a1ba2231876f89fcc1dc5a7
> 1..6
> # Start of mount tests
> ok 1 /mount/create_mount_point
> ok 2 /mount/mount
> ok 3 /mount/umount
> # GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
> ok 4 /mount/device_mounted_false
> **
> ERROR:../tests/mount.c:110:test_device_mounted_true: 'pu_umount(mount_point, &fixture->error)' should be TRUE
> not ok /mount/device_mounted_true - ERROR:../tests/mount.c:110:test_device_mounted_true: 'pu_umount(mount_point, &fixture->error)' should be TRUE
> Bail out!
> ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
> 1/1 partup:root / mount FAIL            0.32s   killed by signal 6 SIGABRT
> ```
I assume that mounting and immediately unmounting is a timing problem here. I'll try waiting for the mount operation to be finished.
> 
> Also, `tests/data/root.ext4` gets modified after running the unit tests. This should be prevented. Maybe the change comes through the mount/umount operation? Looking at the hexdump of `tests/data/root.ext4` before and after running the tests did not reveal any obvious things, though. thinking
I'll switch to using a copy of `root.ext` although I don't think this has anything to do with the failing tests.

**Comment by landerweit-phytec** [comment]
A copy of `data/root.ext4` is now used to not alter the original file.

File: tests/helper.h [hunk 4]
```diff
@@ -45,5 +53,9 @@ void package_files_setup(PackageFilesFixture *fixture,
                          gconstpointer user_data);
 void package_files_teardown(PackageFilesFixture *fixture,
                             gconstpointer user_data);
+void file_copy_setup(FileCopyFixture *fixture,
+                     gconstpointer filename);
+void file_copy_teardown(FileCopyFixture *fixture,
+                        gconstpointer user_data);
```
Comment (mschwan-phytec) [inline tests/helper.h:59]: How about keeping this consistent with the other fixtures and rename it to `copy_file_*` and `CopyFileFixture`?

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

**Comment by mschwan-phytec** [comment]
> A copy of `data/root.ext4` is now used to not alter the original file.

Somehow the test still modifies the original file for me. :thinking:

**Review by mschwan-phytec — APPROVED** [review]

**Pull request merged**